common-ui/src/main/java/com/lgtm/android/common_ui/adapter/SduiAdapter.kt

```diff
@@ -5,26 +5,36 @@ import androidx.recyclerview.widget.ListAdapter
 import com.lgtm.android.common_ui.util.ItemDiffCallback
 import com.lgtm.android.common_ui.viewholder.SduiBaseHolder
 import com.lgtm.android.common_ui.viewholder.SduiItemViewHolder
+import com.lgtm.android.common_ui.viewholder.SduiSubItemViewHolder
 import com.lgtm.android.common_ui.viewholder.getSduiViewHolder
 import com.lgtm.domain.entity.response.SduiItemVO
 import com.lgtm.domain.server_drive_ui.SduiContent
 import com.lgtm.domain.server_drive_ui.SduiViewType
 
 
 class SduiAdapter(
-    private val onMissionClickListener: (SduiContent) -> Unit
+    private val onMissionClickListener: (SduiContent) -> Unit,
+    private val onRecommendationClickListener: () -> Unit
 ) : ListAdapter<SduiItemVO, SduiBaseHolder>(
     ItemDiffCallback<SduiItemVO>(onContentsTheSame = { old, new -> old == new },
         onItemsTheSame = { old, new -> old.content == new.content })
 ) {
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SduiBaseHolder {
         val viewHolder = getSduiViewHolder(parent, SduiViewType.getViewTypeByOrdinal(viewType))
-        if (viewType == SduiViewType.ITEM.ordinal)
-            return viewHolder.apply {
-                (viewHolder as SduiItemViewHolder).setNavigateToMissionDetail(onMissionClickListener)
+        return when (viewType) {
+            SduiViewType.ITEM.ordinal -> {
+                viewHolder.apply {
+                    (viewHolder as SduiItemViewHolder).setNavigateToMissionDetail(onMissionClickListener)
+                }
             }
-        return viewHolder
+            SduiViewType.SUBITEM.ordinal -> {
+                viewHolder.apply {
+                    (viewHolder as SduiSubItemViewHolder).setNavigateToRecommendationDashboard(onRecommendationClickListener)
+                }
+            }
+            else -> viewHolder
+        }
```

Comment:
return값 때문에 apply쓰니까 좀 코드가 길어져서 함수 분리 해봤어!!

```suggestion
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SduiBaseHolder {
        val viewHolder = getSduiViewHolder(parent, SduiViewType.getViewTypeByOrdinal(viewType))

        when (SduiViewType.getViewTypeByOrdinal(viewType)) {
            SduiViewType.ITEM -> setOnMissionClickListener(viewHolder)
            SduiViewType.SUBITEM -> setOnRecommendationClickListener(viewHolder)
            else -> Unit
        }

        return viewHolder
    }

    private fun setOnMissionClickListener(viewHolder: SduiBaseHolder) {
        viewHolder as SduiItemViewHolder
        viewHolder.setNavigateToMissionDetail(onMissionClickListener)
    }

    private fun setOnRecommendationClickListener(viewHolder: SduiBaseHolder) {
        viewHolder as SduiSubItemViewHolder
        viewHolder.setNavigateToRecommendationDashboard(onRecommendationClickListener)
    }
```